packages/terra-button/src/Button.scss

```diff
@@ -54,74 +50,86 @@
       padding: 0;
     }
 
-    // Restore the focus styles unset by the previous rule.
-    &:-moz-focusring {
-      outline: 1px dotted #1c1f21;
-      -moz-outline-radius: 0.25em;
+    &:focus {
+      outline: none;
+    }
+
+    &:active {
+      outline: none;
+      -webkit-tap-highlight-color: transparent;
     }
 
     &:disabled,
     &.is-disabled {
       cursor: default;
-      opacity: 0.7;
+      opacity: var(--terra-button-disabled-opacity, 0.3);
       pointer-events: none; // prevent pointer interaction for anchor elements
     }
   }
 
-  // Button modifiers
-  .default {
-    @include terra-button-color($terra-button-default-color-scheme);
+  // Button variant modifiers
+  .neutral {
+    @include terra-button-variant-color($terra-button-neutral-color-scheme);
   }
 
-  .primary {
-    @include terra-button-color($terra-button-primary-color-scheme);
+  .emphasis {
+    @include terra-button-variant-color($terra-button-emphasis-color-scheme);
   }
 
-  .secondary {
-    @include terra-button-secondary($terra-button-secondary-color-scheme);
+  .de-emphasis {
+    @include terra-button-variant-color($terra-button-de-emphasis-color-scheme);
   }
 
-  .link {
-    @include terra-button-link($terra-button-link-color-scheme);
+  .action {
+    @include terra-button-variant-color($terra-button-action-color-scheme);
+    border-radius: var(--terra-button-action-border-radius, 0.2142rem);
   }
 
-  // Button size modifiers
-  .tiny {
-    font-size: map-get($terra-button-font-sizes, tiny);
+  .utility {
+    @include terra-button-variant-color($terra-button-utility-color-scheme);
+    border-radius: var(--terra-button-utility-border-radius, 0.2142rem);
   }
 
-  .small {
-    font-size: map-get($terra-button-font-sizes, small);
+  // Button style modifiers
+  .block {
+    display: block;
+    width: 100%;
   }
 
-  .medium {
-    font-size: map-get($terra-button-font-sizes, medium);
+  .text-only {
+    margin-left: var(--terra-button-text-only-margin-left, 0.714rem);
+    margin-right: var(--terra-button-text-only-margin-right, 0.714rem);
   }
 
-  .large {
-    font-size: map-get($terra-button-font-sizes, large);
+  .text-left {
+    margin-left: var(--terra-button-text-left-margin-left, 0.714rem);
+    margin-right: var(--terra-button-text-left-margin-right, 0.714rem);
   }
 
-  .huge {
-    font-size: map-get($terra-button-font-sizes, huge);
+  .text-right {
+    margin-right: var(--terra-button-text-right-margin-right, 0.714rem);
   }
 
-  // Button style modifiers
-  .block {
-    display: block;
-    width: 100%;
+  .icon {
+    display: inline-block;
+
+    > svg {
+      height: var(--terra-button-icon-height, 1rem);
+      width: var(--terra-button-icon-width, 1rem);
+    }
   }
 
-  .compact {
-    padding-bottom: var(--terra-button-padding-bottom-compact, 0);
-    padding-top: var(--terra-button-padding-top-compact, 0);
+  .icon-only {
```

Comment:
Is it possible to make the CSS smart enough that we don't need these custom props overrides? These icon-only props are confusing IMO